cogs/chal/chal.py

```diff
@@ -1,167 +1,66 @@
+        if (category == "unsolved"):
+            await ctx.send(":x: Unsolved cannot be a category.")
+            return
+        for post in forum.get_posts(exclude_archived=False):
+            if ((post.name or "") == name):
+                await ctx.send(f":x: Challenge {name} already exists.")
                 return
-        await ctx.guild.create_text_channel(
-            name=channel_name,
-            category=category_folder,
-        )
-
-        # Get bot channel and send embed
-
-        embed = discord.Embed(
-            title="🔔 New challenge created!",
-            description=(
-                f"**Challenge name:** {name}\n"
-            ),
-            colour=discord.Colour.dark_gold(),
-        ).set_footer(text=datetime.strftime(datetime.now(), "%a, %d %B %Y, %H:%M UTC"))
-        await ctx.send(embed=embed)
-
-    @subcommand_decorator(flag={'description': "The flag for this challenge."})
-    @run_from_ctf
-    async def solve(self, ctx: SlashContext, flag: str) -> None:
-        """Solves a CTF challenges
-
-        """
-        await ctx.defer()
-        challenge = ctx.channel
 
-        if challenge.name in AUTOGENERATED_CHANNELS.values():
-            await challenge.send(f'You cannot solve the channel {challenge.name}!')
+        # check valid category
+        for tag in forum.available_tags:
+            if (tag.name == category):
+                break
+        else:  # this indentation is intentional
+            await ctx.send(f":x: Could not find category {category}.")
             return
-        elif challenge.name.startswith('✔️'):
-            await challenge.send(f'This challenge was already solved!')
-            return
-
-        embed = discord.Embed(
-            title=f":white_check_mark: '{challenge.name}' was solved with the flag ||{flag}||"
-        )
-        await challenge.edit(name='✔️ ' + challenge.name)
-        await ctx.send('Marked channel as solved.')
-        logs_channel = discord.utils.get(
-            challenge.category.text_channels, name=AUTOGENERATED_CHANNELS['logs'])
 
-        await logs_channel.send(embed=embed)
+        await forum.create_post(name=name, content=name, applied_tags=[tag, "unsolved"])
+        await ctx.send(f"Created {name}.")
 
-        # TODO ctfd integration
-        # raise NotImplementedError()
-
-    @commands.bot_has_permissions(manage_channels=True)
-    @commands.has_any_role(HELPER_ROLE_ID, ADMIN_ROLE_ID)
-    @subcommand_decorator()
-    @run_from_ctf
-    async def delete(self, ctx: SlashContext) -> None:
-        """
-        Deletes a CTF challenge channel
-        """
-        challenge = ctx.channel
-
-        if challenge.name in AUTOGENERATED_CHANNELS.values():
-            await challenge.send(f'You cannot delete the channel {challenge.name}!')
+    @create.autocomplete("category")
+    async def get_categories(self, ctx: interactions.AutocompleteContext):
```

Comment:
What is the purpose of this command?

Reply:
This is the autocomplete that provides the list of challenge categories for the /chal create command. I have added a docstring to reflect that.

Reply:
Makes sense